### PR TITLE
Changes in VB Blocks & Bug Fixes & Translation

### PR DESCRIPTION
**Pull request by Sal4iDev**
- Added `Deep` in `[Config] Config Keys`
- Changed `ExpressionParameter` with label `GUI` in `[Player] Open GUI` to `InputParameter` with label `GUI ID`

### CODE REVIEW

**Comment by OfficialDonut** [comment]
- Put parameters on the same line in `new Block(...)` so the style is consistent across all classes.
- The GUI ID field is more flexible as an ExpressionParameter. Someone may want to use an arbitrary string expression as the ID e.g. a command argument.

**Comment by Sal4iDev** [comment]
however, in the Create GUI component, in the GUI ID field, InputParameter is set. Then you should add an ExpressionParameter there so that you can also use different values ​​there?

**Comment by Sal4iDev** [comment]
> however, in the Create GUI component, in the GUI ID field, InputParameter is set. Then you should add an ExpressionParameter there so that you can also use different values ​​there?

although I still don't see the point

**Comment by OfficialDonut** [comment]
It makes sense for the ID to be hardcoded when defining the GUI, but forcing the ID to be hardcoded when opening is unnecessarily restrictive.

A contrived example:
```
command
  openGui(arg0)
```
vs
```
command
  if arg0 == "gui1"
    openGui(gui1)
  else if arg0 == "gui2"
    openGui(gui2)
  else if arg0 == "gui3"
    openGui(gui3)
```

**Comment by Sal4iDev** [comment]
does it throw an error if the id is not valid? then you need to do this:

```
command
  attempt execution 
    openGui(arg0)
```

okay.

**Comment by OfficialDonut** [comment]
Yes it would throw an exception if the id was invalid. Attempt execution is an option but letting an exception get thrown isn't best practice, instead you could do input validation by having a list of GUI IDs and checking if the list contains it.

**Comment by Sal4iDev** [comment]
why is it better to check through the list?

**Comment by OfficialDonut** [comment]
Explicitly checking things better conveys your intent and you dont have to rely on the implementation of the open gui code e.g. what if the code changes so it silently fails instead of throwing an exception?

**Comment by Sal4iDev** [comment]
image
Done. Also now `Stop Loop` will work in `Advanced Number Loop`

**Comment by Sal4iDev** [comment]
And `Continue Loop` will work too

File: VB-Application/src/main/java/com/gmail/visualbukkit/blocks/definitions/ExprPluginAuthors.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package com.gmail.visualbukkit.blocks.definitions;
+
+import com.gmail.visualbukkit.blocks.ClassInfo;
+import com.gmail.visualbukkit.blocks.Expression;
+
+public class ExprPluginAuthors extends Expression {
+
+    public ExprPluginAuthors() {
+        super("expr-plugin-authors", "Plugin Authors", "Bukkit", "Authors of the plugin");
```
Comment (OfficialDonut) [inline VB-Application/src/main/java/com/gmail/visualbukkit/blocks/definitions/ExprPluginAuthors.java]: For consistency, "The authors of the plugin"

File: VB-Application/src/main/java/com/gmail/visualbukkit/blocks/definitions/StatContinueLoop.java [hunk 2]
```diff
@@ -14,7 +14,7 @@ public Block createBlock() {
             @Override
             public void update() {
                 super.update();
-                setValid(checkForContainer(StatListLoop.class) || checkForContainer(StatNumberLoop.class) || checkForContainer(StatNumberLoop.class));
+                setValid(checkForContainer(StatListLoop.class) || checkForContainer(StatNumberLoop.class) || checkForContainer(StatAdvancedNumberLoop.class));
```
Comment (OfficialDonut) [inline VB-Application/src/main/java/com/gmail/visualbukkit/blocks/definitions/StatContinueLoop.java]: Please add `checkForContainer(StatWhileLoop.class)`

Comment (Sal4iDev) [reply VB-Application/src/main/java/com/gmail/visualbukkit/blocks/definitions/StatContinueLoop.java]: sure

**Comment by Sal4iDev** [comment]
Done

**Comment by Sal4iDev** [comment]
- Added `(s)` to `label.plugin_author`

**Comment by Sal4iDev** [comment]
I also added the ability to select the `<`, `<=`, `>`, `>=` mode in the `Advanced Number Loop`

File: VB-Application/src/main/resources/lang/GUI.properties [hunk 3]
```diff
@@ -55,7 +55,7 @@ label.enabled_extensions=Enabled Extensions (for current project)
 label.events=Events
 label.installed_extensions=Installed Extensions
 label.other=Other
-label.plugin_author=Author:
+label.plugin_author=Author(s):
```
Comment (OfficialDonut) [inline VB-Application/src/main/resources/lang/GUI.properties:58]: https://github.com/OfficialDonut/VisualBukkit/blob/5f891442b066fe850b42c9f18b2be22cfc27ed20/VB-Application/src/main/java/com/gmail/visualbukkit/project/PluginBuilder.java#L232
This has to be changed to support multiple authors

Comment (Sal4iDev) [reply VB-Application/src/main/resources/lang/GUI.properties:58]: Oh, yeah, sure

Comment (Sal4iDev) [reply VB-Application/src/main/resources/lang/GUI.properties:58]: done

**Pull request merged**